servant-client/src/Servant/Client/PerformRequest/Base.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ViewPatterns #-}
+
+{-# OPTIONS_GHC -fno-warn-name-shadowing #-}
+
+module Servant.Client.PerformRequest.Base where
+
+import           Control.Arrow
+import           Control.Exception
+import qualified Data.ByteString.Char8 as BS
+import           Data.ByteString.Lazy
+import           Data.CaseInsensitive
+import           Data.Char
+import           Data.String.Conversions
+import           Data.Typeable
+import           Network.HTTP.Media
+import           Network.HTTP.Types
+
+data ServantError
+  = FailureResponse
+    { responseStatus            :: Status
+    , responseContentType       :: MediaType
+    , responseBody              :: ByteString
+    }
+  | DecodeFailure
+    { decodeError               :: String
+    , responseContentType       :: MediaType
+    , responseBody              :: ByteString
+    }
+  | UnsupportedContentType
+    { responseContentType       :: MediaType
+    , responseBody              :: ByteString
+    }
+  | InvalidContentTypeHeader
+    { responseContentTypeHeader :: ByteString
+    , responseBody              :: ByteString
+    }
+  | ConnectionError
+    { connectionError           :: SomeException
+    }
+  deriving (Show, Typeable)
+
+instance Exception ServantError
+
+parseHeaders :: String -> ResponseHeaders
```

Comment:
Yes, you got me. ;)

I thought since it's just a pure function without additional dependencies I don't have to bother to CPP-exclude it. Should I put it into the ghcjs-specific module?